src/Checkbox.js

```diff
@@ -38,7 +52,7 @@ class Checkbox extends React.Component {
                     className="mdl-checkbox__input"
                     { ...inputProps }
                 />
-                {label && <span className="mdl-checkbox__label">{label}</span>}
+                {this.labelElement()}
```

Comment:
I'd keep the old syntax but use something like `{label && this.formatLabel(label)}`

With `formatLabel` returing the label inside the mdl element if it's a string, or just returning `label` for the others cases. It will simplify the function also since checking the object is not needed.